aws/rakefiles/deploy.rake

```diff
@@ -115,10 +109,21 @@ task :install_charts => [:configure_kubectl, :generate_modules, :setup_system_co
   installed_charts = installed_charts.split("\n")
   @gpii_helmcharts.each do |chart|
     chart_config = YAML.load_file("#{@tmpdir}-modules/deploy/helms/#{chart}/custom-values.yaml")
+    # Extracting chart name and namespace from chart-metadata
+    if chart_config['chart-metadata'] && chart_config['chart-metadata']['name']
+      chart_name = chart_config['chart-metadata']['name']
+    else
+      chart_name = chart
+    end
+    if chart_config['chart-metadata'] && chart_config['chart-metadata']['namespace']
+      chart_namespace = chart_config['chart-metadata']['namespace']
+    else
+      chart_namespace = 'default'
+    end
     if installed_charts.include?(chart)
       begin
         wait_for(
-          "helm upgrade --recreate-pods -f #{@tmpdir}-modules/deploy/helms/#{chart}/custom-values.yaml #{chart_config['helm-override'] && chart_config['helm-override']['name'] ? chart_config['helm-override']['name'] : chart} #{@tmpdir}-modules/deploy/helms/#{chart}",
+          "helm upgrade --recreate-pods -f #{@tmpdir}-modules/deploy/helms/#{chart}/custom-values.yaml #{chart_name} #{@tmpdir}-modules/deploy/helms/#{chart}",
```

Comment:
Can we include --namespace here for consistency with the `install` path?

Reply:
Done